### ISSUE TITLE
Is the project dead?

### ISSUE THREAD

**Issue by TamasSzerb** [gitblit/gitblit#1265]
Hello,

I haven't seen new release for long time, so would be great to know the plans with this.

Thanks,

Tamas

**Comment by paulsputer** [comment]
Hi Tamas, certainly not dead from my perspective, I use it daily.  

I would interpret the no release for a long time as a sign of stability, which for something as important as storing your code and other assets is valuable.

Saying that there are many features I had hoped to get around to adding but I simply don't have the time these days.

**Comment by pingunaut** [comment]
Hi all,
there are pull requests, that were opened more than one year ago, so my opinion is: 
yes, stability is good, but a little speed up in development would be a good thing. 
As far as I can see, #725 is the only open issue in milestone 1.9.0 that has not been addressed yet. 
If no one started one that one, i'd like to give it a try, so 1.9.0 can finally be released. 

Just my 2cents,
Best regards,
Martin

**Comment by rnveach** [comment]
> I would interpret the no release for a long time as a sign of stability

I disagree by the number of bug reports and exceptions people have reported that have not been fixed.
You can't say an application is stable if it is failing all over the place.

> were opened more than one year ago

I take this as a sign as the maintainer(s) don't want to be very active or engage the project.
The oldest PR has no push back for changes or reasoning why it can't be merged, so it just hasn't been looked at (as far as I know) which is sad since it is more than a year old.
Contribution graph definitely shows signs of slowing and last commit added was 6 months ago. https://github.com/gitblit/gitblit/graphs/contributors

**Comment by silvernode** [comment]
This is probably a hobby for the founder anyway and he probably has a family so I think I am just going to calmly wait for the next release. With that said the current versions works just fine anyway so until I start noticing breakage from lagging behind in development I can keep my cool.

**Comment by fzs** [comment]
I certainly wouldn't call Gitblit dead. But I agree that it is not getting enough attention right now.
As for me, that is due to having moved houses last year and still setting up in the new home and at the same time work being crazy. Which is a bummer, as I am still very interested in Gitblit. I don't know how James pulled it off in the first place.

As for updates, I do actually see Gitblit more as a tool used in a professional setting than a private one. At least that is how I use it. So I would rate stability over adding new features in a high rate. Again, that should include fixing actual bugs, agreed.

That being said, what I see as a major obstacle right now is that state of the tests. There are hardly any unit tests, a lack in integration tests, the issue with external repos being used, which aren't available anymore, and so on. I consider a solid base of unit and integration tests a necessary prerequisite to adding fixes and new features while keeping the project stable. While James may be able to know every impact of a change, as he knows the project inside out, that is not true for other contributors. So if anyone feels like helping out with enabling a speed-up in development on Gitblit again, I can only suggest to tackle the daunting and boring task of fixing the tests and increasing test coverage.

**Comment by fzs** [comment]
Or, if you are so inclined, create a/some git repositories with trees that cover all sorts of test cases, that can be used as local repositories for the test, so that the remote ones can be replaced. Using remote repos not under project control for tests wasn't the best idea anyways.

**Comment by goettl79** [comment]
Summarize with #1274 the following would be nice to have

1.  a more standardized build environment (no more ant + moxie)
2. stable tests, in one repository
3. a person accepts pull requests 

All tasks are above the timelimit I'd be able to spare. However, one quick win would be to setup a continuous docker build on dockerhub. How about accepting the simple pull request "#1232"?

**Comment by wbrawner** [comment]
I'd love to see this project pick up speed in development, so I'll write tests to help improve stability. But before I go submitting PRs, is there anyone with write access who would be willing and able to review and merge them in?

**Comment by fzs** [comment]
Hi Billy,
I'd be happy to merge them.
Feel free to tackle whatever you feel most comfortable with or see the most benefit in.
I thought would probably be to add tests for current bug reports (which will of course fail) so that they can prove a bug fixed, or also the problem of adding git repositories to the project which can be used for tests, as the ones that were used are gone.

**Comment by wbrawner** [comment]
Fantastic, I'll take a look around then and see what I can do :)

**Comment by dpursehouse** [comment]
Are there any plans to release 1.9.0 soon?  At the gerrit project, we're building the head of gitblit master (1.9.0-SNAPSHOT) in CI to include in the gitblit plugin. It would be nice if we could just download a released 1.9.0 artifact instead.

**Comment by johnstile** [comment]
Is there a timeline to complete the outstanding issues for 1.9.0?
A time line goal would be very encouraging for this project.

**Comment by wbrawner** [comment]
Unfortunately no work will be done as long as the tests are broken. I've started working on the tests but it's been a bit of a crazy year for me personally so I haven't had much free time. I'm expecting to finally have some downtime for the next two weeks though so that I can take a look and attempt to fix the tests and get feature development going again. The problem is that it's not a straightforward task and like I previously mentioned, I haven't had much time this year
